enterprise/migrations/0136_auto_20210629_2129.py

```diff
@@ -0,0 +1,70 @@
+# Generated by Django 2.2.20 on 2021-06-29 21:29
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+import model_utils.fields
+import simple_history.models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('enterprise', '0135_adminnotification_adminnotificationfilter_adminnotificationread'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='systemwideenterpriseuserroleassignment',
+            name='role',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='system_wide_role_assignments', to='enterprise.SystemWideEnterpriseRole'),
+        ),
+        migrations.CreateModel(
+            name='HistoricalSystemWideEnterpriseUserRoleAssignment',
```

Comment:
Do we add in any additional tests for history models (doubtful)? and is there a way you could mimic the old behavior in a test to prove in this PR that the changes will fix it?

Reply:
To answer your first question: We don't have any existing tests. It'd probably be possible to verify the model gets instantiated when the original is saved, but mostly we'd be testing that simple history does what it's supposed to and that's what the tests they presumably have in their own repo should be doing. 😄 

For the second: Not really. For a few reasons:
- Now that we have a history table, we can't really mimic what it was like without it.
- Tests that look for the absence of something are often pretty problematic as I tend to question if it's correctly not finding the object or if it's just that test conditions wouldn't have generated it anyways (this is especially true as you get into mocks and frontend code. I have many, many times found my test looking for an absence is only functioning because the positive case is also not working 😞 ). You typically want to format your tests in a way that they can look for something's existence (usually we look for exceptions and I guess I could custom make one to detect this but that might get pretty complicated)
- [extremely personal opinion]: I don't like adding tests to detect an issue that I've already fixed. Unless it's a major issue (ie if we pushed code that would have otherwise failed this new test and it breaks production), or it's a particular utility/method/library that is used in a variety of places, a hyper specific test just adds time to our build process for no particular benefit.